light-client/src/consensus/rpc/portal_rpc.rs

```diff
@@ -0,0 +1,316 @@
+                    error = %err,
+                    "Error receiving FindContent query response"
+                );
+                return Err(err.into());
+            }
+        }
+    }
+
+    async fn get_optimistic_update(&self) -> eyre::Result<LightClientOptimisticUpdateCapella> {
+        let expected_current_slot = expected_current_slot();
+        let optimistic_update_key =
+            BeaconContentKey::LightClientOptimisticUpdate(LightClientOptimisticUpdateKey {
+                signature_slot: expected_current_slot,
+            });
+
+        let (tx, rx) = oneshot::channel();
+
+        let overlay_command = OverlayCommand::FindContentQuery {
+            target: optimistic_update_key,
+            callback: tx,
+            is_trace: false,
+        };
+
+        if let Err(err) = self.overlay_tx.send(overlay_command) {
+            warn!(
+                protocol = "beacon",
+                error = %err,
+                "Error submitting FindContent query to service"
+            );
+            return Err(err.into());
+        }
+
+        match rx.await {
+            Ok(result) => {
+                let optimistic_update = BeaconContentValue::decode(
+                    &result
+                        .0
+                        .ok_or(format!("LightClientOptimisticUpdate content with signature slot {expected_current_slot} not found on the network."))
+                        .map_err(|err| eyre!("{err}"))?,
+                )?;
+
+                if let BeaconContentValue::LightClientOptimisticUpdate(optimistic_update) =
+                    optimistic_update
+                {
+                    if let LightClientOptimisticUpdate::Capella(optimistic_update_capella) =
+                        optimistic_update.update
+                    {
+                        Ok(optimistic_update_capella)
+                    } else {
+                        return Err(eyre!(
+                            "Error converting LightClientOptimisticUpdate to LightClientOptimisticUpdateCapella"
+                        ));
+                    }
+                } else {
+                    return Err(eyre!(
+                        "Error converting BeaconContentValue to LightClientOptimisticUpdate"
+                    ));
+                }
+            }
+            Err(err) => {
+                warn!(
+                    protocol = %"beacon",
+                    error = %err,
+                    "Error receiving FindContent query response"
+                );
+                return Err(err.into());
+            }
+        }
+    }
+
+    async fn chain_id(&self) -> eyre::Result<u64> {
+        Ok(1)
+    }
+
+    fn name(&self) -> String {
+        "portal".to_string()
+    }
+}
+
+pub fn expected_current_slot() -> u64 {
+    let now = SystemTime::now();
+    let now = now.duration_since(UNIX_EPOCH).expect("Time went backwards");
+    let since_genesis = now - std::time::Duration::from_secs(BEACON_GENESIS_TIME);
```

Comment:
Would like to see this import moved to the top. If it makes sense, I think you can even move the import to the const definition. eg. `const RETRY_AFTER: Duration = Duration::from_secs(15);`